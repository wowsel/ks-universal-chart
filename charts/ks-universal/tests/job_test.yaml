suite: test job
templates:
  - job.yaml
tests:
  - it: should create a job with specified configuration
    set:
      jobs:
        test-migration:
          parallelism: 1
          completions: 1
          backoffLimit: 3
          containers:
            migration:
              image: migration-tool
              imageTag: "1.0.0"
              command: 
                - /bin/sh
                - -c
                - migrate.sh
              env:
                - name: DB_URL
                  value: postgres://user:password@db:5432/app
    asserts:
      - isKind:
          of: Job
      - isAPIVersion:
          of: batch/v1
      - equal:
          path: metadata.name
          value: test-migration
      - equal:
          path: spec.parallelism
          value: 1
      - equal:
          path: spec.completions
          value: 1
      - equal:
          path: spec.backoffLimit
          value: 3
      - matchSnapshot:
          path: spec.template.spec 