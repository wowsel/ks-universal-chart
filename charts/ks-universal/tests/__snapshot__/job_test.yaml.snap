should create a job with specified configuration:
  1: |
    containers:
      - command:
          - /bin/sh
          - -c
          - migrate.sh
        env:
          - name: DB_URL
            value: postgres://user:password@db:5432/app
        image: migration-tool:1.0.0
        name: migration
    restartPolicy: Never
